From 17b0edccd25b16ea4e680d55465a89160770afbf Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:39:24 -0400
Subject: [alsa] pcm: Add tstamp_type and proto to sw_params compat layer

Message-id: <140972823573-396-patchset-jkysela@redhat.com>
Patchwork-id: 91092
O-Subject: [RHEL 7.1 PATCH 395/442] ALSA: pcm: Add tstamp_type and proto to sw_params compat layer
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

I forgot to add the new fields in sw_params to 32bit compat layer.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: e58c295c041d17aa4298c1c8f618752f05d5c6ee)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/core/pcm_compat.c b/sound/core/pcm_compat.c
index af49721..102e8fd 100644
--- a/sound/core/pcm_compat.c
+++ b/sound/core/pcm_compat.c
@@ -101,7 +101,9 @@ struct snd_pcm_sw_params32 {
  u32 silence_threshold;
  u32 silence_size;
  u32 boundary;
- unsigned char reserved[64];
+ u32 proto;
+ u32 tstamp_type;
+ unsigned char reserved[56];
 };
 
 /* recalcuate the boundary within 32bit */
@@ -133,7 +135,9 @@ static int snd_pcm_ioctl_sw_params_compat(struct snd_pcm_substream *substream,
      get_user(params.start_threshold, &src->start_threshold) ||
      get_user(params.stop_threshold, &src->stop_threshold) ||
      get_user(params.silence_threshold, &src->silence_threshold) ||
-     get_user(params.silence_size, &src->silence_size))
+     get_user(params.silence_size, &src->silence_size) ||
+     get_user(params.tstamp_type, &src->tstamp_type) ||
+     get_user(params.proto, &src->proto))
   return -EFAULT;
  /*
   * Check silent_size parameter.  Since we have 64bit boundary,
-- 
1.7.1